shader/main3d.vert.glsl

```diff
@@ -5,17 +5,27 @@ void main()
   vec3 norm = inNormal;
   vec3 normScreen = normalize(matNorm * norm);
 
+  cc_shade = inColor;
+  flags = inFlags;
+  geoMode = inGeoMode;
+  othermodeL = inOthermodeL;
+  othermodeH = inOthermodeH;
+
   vec4 lightTotal = vec4(ccData.ambientColor.rgb, 0.0);
   for(int i=0; i<2; ++i) {
     float lightStren = max(dot(norm, ccData.lightDir[i].xyz), 0.0);
     lightTotal += ccData.lightColor[i] * (lightStren * 2);
   }
-  
-  // Ambient light
-  cc_shade.rgb = linearToGamma(lightTotal.rgb);
-  cc_shade.a = 1.0;
-  cc_shade *= inColor;
+
+  lightTotal.rgb = linearToGamma(clamp(lightTotal.rgb, 0.0, 1.0));
+
+#if defined(F3DEX3)
```

Comment:
This can be removed to just keep the EX3 path in general. python (fast64 or material parser) should make sure packed normals are never set in non  EX3